cl/search/tests.py

```diff
@@ -2537,6 +2537,235 @@ async def test_pa_search_form_search_and_filtering(self) -> None:
             msg="'Riley' should come before 'Peck' and before 'Smith' when order_by asc.",
         )
 
+    def test_keep_in_sync_related_pa_objects(self) -> None:
```

Comment:
This is a long test with a lot of IO. Would it be possible to make it into at least two or maybe three or more? When django runs tests in parallel, long tests can tie up a CPU and slow things down while shorter ones can be parallelized.

It's OK if breaking this up doesn't make sense, but if there's an easy and intuitive way to do so, we probably should.

Reply:
Indeed, it's a good idea to split the ES tests, including dividing them into different files. This is something Eduardo and I have been discussing in order to make search tests more maintainable, so we can work on different modules/documents simultaneously.

So, I believe this is a great opportunity to start splitting tests, since we'll update the ES open PRs with this code once it's merged.

The new structure for search tests is:

- search
  - tests
    - __init__.py
    - tests.py
    - tests_es_parenthetical.py
    
`tests.py` is the same file we had only `ElasticSearchTest` class and its dependencies were removed.

`tests_es_parenthetical.py`, include `ElasticSearchTest` (renamed to ParentheticalESTest) and created a new class `ParentheticalESSignalProcessorTest` for the `ESSignalProcessorTest` tests  split into 4 test methods.

There were some complications doing this ES test split.

- When running tests in `parallel` test methods for different class in the same file can run in parallel.
- The problem of running test methods concurrently from test classes that populate the same elasticsearch index is that it creates conflicts on the shared index, including documents mismatch that lead tests to fail.
- This problem can't be prevented using `lockfile = __file__` since it only works to avoid running test classes concurrently when they're in different files.
- In order to solve the problem is required to ES isolate indices on every test class. I tried different methods but the most simpler is to use a different index name on each test class, so documents for the same type are populated in different indices, this is a single line that needs to be added before populating the ES index:
`parenthetical_group_index._name = cls.__name__.lower()`, in this example `parenthetical_group_index` is the index object on `documents.py`
ChatGPT
Here's the corrected version:

There were some complications doing this test split:

- When running tests in `parallel`, test methods from different classes in the same file can run concurrently.
- The issue with running test methods concurrently from classes that populate the same Elasticsearch index is that it creates conflicts on the shared index. This includes document mismatches that cause tests to fail.
- This problem cannot be prevented using `lockfile = __file__` as it only works to avoid running test classes concurrently when they're in different files.
- To solve this problem, it's necessary to isolate the ES indices for every test class. I tried different methods, but the simplest one is to use a different index name for each test class. That way, documents of the same type are populated in different indices. This requires a single line to be added before populating the ES index: parenthetical_group_index._name = cls.__name__.lower(). In this example, `parenthetical_group_index` is the index object in `documents.py`.

Reply:
ChatGPT has entered the chatroom? What's going on here? ")

Reply:
haha sorry I use it to correct my grammar 😅

Reply:
Wild world. Anyway, sounds good.